controllers/argocd/deployment.go

```diff
@@ -533,6 +533,9 @@ func (r *ReconcileArgoCD) reconcileRedisDeployment(cr *argoproj.ArgoCD, useTLS b
 			// Deployment exists but component enabled flag has been set to false, delete the Deployment
 			log.Info("Redis exists but should be disabled. Deleting existing redis.")
 			return r.Client.Delete(context.TODO(), deploy)
+		} else if cr.Spec.Redis.Remote != nil && *cr.Spec.Redis.Remote != "" {
```

Comment:
Could we define a new `func IsRemote()`, similar to `IsEnabled()`, for this check? This would enhance code readability.
https://github.com/argoproj-labs/argocd-operator/blob/603addacfb9d4c00e33d13bc561374b8a041366b/api/v1beta1/argocd_types.go#L464-L466